cmd/snap-confine/mount-support.c

```diff
@@ -404,6 +404,64 @@ static void sc_bootstrap_mount_namespace(const struct sc_mount_config *config)
 			    dst);
 		}
 	}
+	// Since we mounted /etc from the host filesystem the /etc/os-release file
+	// (which may not be present) may be a symbolic link. It is expected that
+	// we see the real /etc/os-release file so let's do our best to show
+	// /etc/os-release as it looks like on the classic distribution.
+	//
+	// If the symbolic link in /etc is pointing to /usr/lib/os-release then
+	// bind mount /usr/lib/os-release over itself in the core snap. This way
+	// the symlink (which we cannot be the target of a bind mount as the kernel
+	// will always follow the link when performing the mount operation) will
+	// work as expected.
+	//
+	// https://bugs.launchpad.net/canonical-livepatch-client/+bug/1667470
```

Comment:
Agree with @mvo5 and @jdstrand.. the whole point of these files is to give an idea to the application what it can expect from the environment around it. I can only imagine pain out of pretending to be in the outer world when the local environment is completely unlike it.